### PR TITLE
ROE-584 adding Matomo tracking and cookie-consent banner

### PR DESCRIPTION
**Pull request by SVarallo**
### JIRA link
https://companieshouse.atlassian.net/browse/ROE-584

### Change description
Work done to add the piwik and cookie consent scripts to service. This will also need docker and chs-config repo work to add the PIWIK_SITE_ID and PIWIK_URL variables to the different environments.

### Work checklist

- [ ] Tests added where applicable
- [x] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/controllers/confirmation.controller.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { logger } from "../utils/logger";
 import { CONFIRMATION_PAGE } from "../config";
 import { getLoggedInUserEmail } from "../utils/session";
 import { getApplicationData } from "../utils/application.data";
-import { ApplicationData } from "../model/application.model";
+import { ApplicationData } from "../model";
```
Comment (markpit) [inline src/controllers/confirmation.controller.ts:7]: i think the import will still work as ApplicationData is exported from model/index.ts but seems out of step with our other imports. Is it ok to change this import back?

**Review by mwestacott — COMMENTED** [review]
Apart from @markpit 's comment on the imports this looks fine to me.

File: src/controllers/confirmation.controller.ts (hunk 1 above)
Comment (SVarallo) [reply src/controllers/confirmation.controller.ts:7]: Sure, it's just something my IDE flagged to simplify it. I'll turn it back

File: src/controllers/beneficial.owner.individual.controller.ts [hunk 2]
```diff
@@ -53,6 +53,7 @@ export const getByID = (req: Request, res: Response, next: NextFunction) => {
 
     return res.render(BENEFICIAL_OWNER_INDIVIDUAL_PAGE, {
       backLinkUrl: BENEFICIAL_OWNER_TYPE_URL,
+      templateName: BENEFICIAL_OWNER_INDIVIDUAL_PAGE,
```
Comment (markpit) [inline src/controllers/beneficial.owner.individual.controller.ts:56]: need to add the template name into the get function as well for the BO individual controller

File: views/interrupt-card.html [hunk 3]
```diff
@@ -27,14 +27,11 @@ <h2 class="govuk-heading-m">Trusts</h2>
...
-          href: "/register-an-overseas-entity/presenter",
-          attributes: {
-            "id": "continue"
-          }
-        }) }}
+
+        <a href="/register-an-overseas-entity/presenter"  id="continue" onclick="_paq.push(['trackGoal', '{{PIWIK_START_GOAL_ID}}'])" role="button" draggable="false" class="govuk-button" data-module="govuk-button">
+          Continue
+        </a>
+
```
Comment (markpit) [inline views/interrupt-card.html:34]: for the button, have you tried adding the extra bits as attributes
{{ govukButton({
          text: "Continue",
          href: "/register-an-overseas-entity/presenter",
          attributes: {
            "id": "continue",
            "onclick": "_paq.push(['trackGoal', '{{PIWIK_START_GOAL_ID}}'])"
          }
        }) }}

maybe add data-module if you need it

Comment (SVarallo) [reply views/interrupt-card.html:34]: Yes I did try that but it wasn't accepting {{PIWIK_START_GOAL_ID}} as a value so had to resort to using the html version

**Review by markpit — APPROVED** [review]

**Pull request merged**